.github/workflows/main.yml

```diff
@@ -13,22 +13,22 @@ concurrency:
   group: ${{ github.ref }}
   cancel-in-progress: true
 
-env:
-  FLUTTER_VERSION:  3.10.6
-  DART_VERSION: 3.0.6
-
 jobs:
-  # because there is no easy way to pass env variables to jobs
+  # because there is no easy way to pass env variables to workflows
   versions:
     runs-on: ubuntu-latest
     outputs:
       flutter_version: ${{ steps.flutterver.outputs.FLUTTER_VERSION }}
       dart_version: ${{ steps.dartver.outputs.DART_VERSION }}
     steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - run: cat .github/workflows/versions.env >> $GITHUB_ENV
       - id: flutterver
-        run: echo "FLUTTER_VERSION=${{ env.FLUTTER_VERSION }}" >> "$GITHUB_OUTPUT"
+        run: echo "FLUTTER_VERSION=${{ env.flutter_version }}" >> "$GITHUB_OUTPUT"
       - id: dartver
-        run: echo "DART_VERSION=${{ env.DART_VERSION }}" >> "$GITHUB_OUTPUT"
+        run: echo "DART_VERSION=${{ env.dart_version }}" >> "$GITHUB_OUTPUT"
```

Comment:
Can you just convert all jobs to use the envfile?

Reply:
yup, just did that. I tried it first then reverted since it was failing and I thought it was a lot of work but yeah now I'm doing it.